src/librustdoc/html/render.rs

```diff
@@ -854,6 +853,35 @@ fn write_shared(cx: &Context,
     }
     try_err!(writeln!(&mut w, "initSearch(searchIndex);"), &dst);
 
+    // Create the directory structure ahead of time
+    let mut item = krate.module.clone().unwrap();
```

Comment:
Is cloning the entire crate here really necessary?

Reply:
It was originally so that i could set the name on the root module, but looking at it again, I can just set the initial path to what i need and let the crate item keep its empty name. Let me try this locally.